lib/acpi/acpi_shell.c

```diff
@@ -220,19 +225,93 @@ static int dump_prt(const struct shell *sh, size_t argc, char **argv)
 static int enum_dev(const struct shell *sh, size_t argc, char **argv)
 {
 	struct acpi_dev *dev;
+	ACPI_RESOURCE *res_lst;
 
-	if (argc < 2) {
+	if (argc < 3) {
+		shell_error(sh, "Invalid arguments [Eg: acpi enum PNP0103 0]");
 		return -EINVAL;
 	}
 
-	dev = acpi_device_get(argv[1], 0);
+	dev = acpi_device_get(argv[1], argv[2]);
 	if (!dev || !dev->res_lst) {
 		shell_error(sh, "acpi get device failed for HID: %s", argv[1]);
 		return -EIO;
 	}
 
 	shell_print(sh, "Name: %s", dev->path ? dev->path : "None");
-	dump_dev_res(sh, dev->res_lst);
+
+	if (dev->path) {
+		if (!acpi_current_resource_get(dev->path, &res_lst)) {
+			dump_dev_res(sh, res_lst);
+			acpi_current_resource_free(res_lst);
+		}
+	}
+
+	return 0;
+}
+
+static int enum_all_dev(const struct shell *sh, size_t argc, char **argv)
+{
+	struct acpi_dev *dev;
+
+	for (int i = 0; i < 1000; i++) {
+		dev = acpi_device_by_index_get(i);
+		if (!dev) {
+			shell_print(sh, "No more ACPI device found!");
+			break;
+		}
+
+		if (!dev->dev_info) {
+			continue;
+		}
+
+		shell_print(sh, "%d) Name:%s, HID:%s", i, dev->path ? dev->path : "Non",
+			    dev->dev_info->HardwareId.String ? dev->dev_info->HardwareId.String
+							     : "");
+	}
+
+	return 0;
+}
+
+static int get_acpi_dev_resource(const struct shell *sh, size_t argc, char **argv)
+{
+	struct acpi_dev *dev;
+	struct acpi_irq_resource irq_res;
+	struct acpi_mmio_resource mmio_res;
+
+	if (argc < 3) {
+		return -EINVAL;
+	}
+
+	dev = acpi_device_get(argv[1], argv[2]);
+	if (!dev) {
+		shell_error(sh, "acpi get device failed for HID: %s", argv[1]);
+		return -EIO;
+	}
+
+	if (dev->path) {
+		shell_print(sh, "Device Path:\n \t%s", dev->path);
+
+		if (!acpi_device_mmio_resource_get(dev, &mmio_res)) {
+
+			shell_print(sh, "Device MMIO resources");
+			for (int i = 0; i < ACPI_RESOURCE_COUNT_GET(&mmio_res); i++) {
+				shell_print(sh, "\tType: %x, Address:%p, Size:%d",
+					ACPI_MULTI_RESOURCE_TYPE_GET(&mmio_res, i),
+					(void *)ACPI_MULTI_MMIO_GET(&mmio_res, i),
+						ACPI_MULTI_RESOURCE_SIZE_GET(&mmio_res, i));
+			}
+		}
+
+		if (!acpi_device_irq_resource_get(dev, &irq_res)) {
+
+			shell_print(sh, "Device IRQ resources");
+			for (int i = 0; i < irq_res.irq_vector_max; i++) {
+				shell_print(sh, "\tIrq Num: %x, Flags:%x", irq_res.irqs[i],
```

Comment:
Same here: missing space after "Flags:". Also, I'd capitalize IRQ for consistency (like you do two lines earlier).